RileyLinkKitTests/PumpOpsSynchronousTests.swift

```diff
@@ -153,87 +145,100 @@ class PumpOpsSynchronousTests: XCTestCase {
+
+        // Debatable whether this should be returned.
         assertArray(timeStampedEvents, containsPumpEvent: squareBolusEventFourHoursBefore)
         XCTAssertTrue(hasMoreEvents)
         XCTAssertFalse(cancelled)
     }
-    
-    func testEventAfterSquareBolusFor523IsNotReturned() {
-        setUpTestWithPumpModel(.Model523)
-        
-        let tempEventBasal = createTempEventBasal2016()
-        let dateComponents = tempEventBasal.timestamp.addingTimeInterval(TimeInterval(hours:-4))
-        let squareBolusEventFourHoursBefore = createSquareBolusEvent(dateComponents: dateComponents)
-        
-        let events:[PumpEvent] = [squareBolusEventFourHoursBefore, tempEventBasal]
-        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
-        
-        // It should not be returned (timestamp from square bolus is valid)
-        assertArray(timeStampedEvents, doesntContainPumpEvent: tempEventBasal)
-    }
-    
-// MARK: Final Square Bolus Event (after Square Bolus)
-    func testSquareBolusAfterDelayedAppendEventFor522IsReturned() {
-        setUpTestWithPumpModel(.Model522)
-        
-        let squareBolus2016 = createSquareBolusEvent2016()
-        let dateComponents = squareBolus2016.timestamp.addingTimeInterval(TimeInterval(hours:-4))
-        let squareBolusEventFourHoursBefore = createSquareBolusEvent(dateComponents: dateComponents)
-        
-        let events:[PumpEvent] = [squareBolusEventFourHoursBefore, squareBolus2016]
-        let (timeStampedEvents, hasMoreEvents, cancelled) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
-        
-        //It should be returned
-        XCTAssertTrue(array(timeStampedEvents, containsPumpEvent: squareBolus2016))
-        XCTAssertTrue(hasMoreEvents)
-        XCTAssertFalse(cancelled)
-    }
-    
-    func testSquareBolusAfterDelayedAppendEventFor523IsNotReturned() {
-        setUpTestWithPumpModel(.Model523)
-        
-        let squareBolus2016 = createSquareBolusEvent2016()
-        let dateComponents = squareBolus2016.timestamp.addingTimeInterval(TimeInterval(hours:-4))
-        let squareBolusEventFourHoursBefore = createSquareBolusEvent(dateComponents: dateComponents)
-        
-        let events:[PumpEvent] = [squareBolusEventFourHoursBefore, squareBolus2016]
-        let (timeStampedEvents, hasMoreEvents, cancelled) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
-        
-        //It should not be returned
-        XCTAssertFalse(array(timeStampedEvents, containsPumpEvent: squareBolus2016))
-        XCTAssertFalse(hasMoreEvents)
-        XCTAssertTrue(cancelled)
-    }
-    
+
+    // Remove or fixme: on a 523 a square wave bolus followed by a another event will be in correct order. So I'm not sure what this is testing.
```

Comment:
I can't tell which test you're talking about here... The diff above doesn't give enough context.